Datagrid/ProxyQuery.php

```diff
@@ -328,6 +328,18 @@ protected function getFixedQueryBuilder(QueryBuilder $queryBuilder)
             $queryBuilderId->addSelect($sortBy);
         }
 
+        // For any orderBy clause defined directly in the dqlParts
+        $dqlParts = $queryBuilderId->getDqlParts();
+        if ($dqlParts['orderBy'] && count($dqlParts['orderBy'])) {
+            $sqlOrderColumns = array();
+            foreach ($dqlParts['orderBy'] as $part) {
+                foreach ($part->getParts() as $orderBy) {
+                    $sqlOrderColumns[] = preg_replace("/\s+(ASC|DESC)$/i", '', $orderBy);
+                }
+            }
+            $queryBuilderId->addSelect(implode(', ', $sqlOrderColumns));
```

Comment:
Why store that in a variable, then add all of them in one go outside the loop? I think you could just do `$queryBuilderId->addSelect(preg_replace("/\s+(ASC|DESC)$/i", '', $orderBy))` on line 337.